services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/rabbitmq.py

```diff
@@ -1,212 +1,83 @@
-        )
-
-    async def _publish_event(self, action: RabbitEventMessageType) -> None:
-        data = EventRabbitMessage(
-            node_id=self._node_id,
-            user_id=self._user_id,
-            project_id=self._project_id,
-            action=action,
-        )
-        assert self._events_exchange  # nosec
-        await self._events_exchange.publish(
-            aio_pika.Message(body=data.json().encode()), routing_key=""
-        )
-
-    async def send_event_reload_iframe(self) -> None:
-        await self._publish_event(action=RabbitEventMessageType.RELOAD_IFRAME)
-
-    async def post_log_message(self, log_msg: str | list[str]) -> None:
-        if isinstance(log_msg, str):
-            log_msg = [log_msg]
-
-        for message in log_msg:
-            await self._channel_queues[self.CHANNEL_LOG].put(message)
-
-    async def close(self) -> None:
-        # wait for queues to be empty before sending the last messages
-        self._keep_running = False
-        if self._queues_worker is not None:
-            self._queues_worker.cancel()
-            with suppress(asyncio.CancelledError):
-                await self._queues_worker
-
-        if self._channel is not None:
-            await self._channel.close()
-        if self._connection is not None:
-            await self._connection.close()
-
-
-async def send_message(rabbitmq: RabbitMQ, msg: str) -> None:
-    log.debug(msg)
-    await rabbitmq.post_log_message(f"[sidecar] {msg}")
+async def post_event_reload_iframe(app: FastAPI) -> None:
+    app_settings: ApplicationSettings = app.state.settings
+    message = EventRabbitMessage(
+        node_id=app_settings.DY_SIDECAR_NODE_ID,
+        user_id=app_settings.DY_SIDECAR_USER_ID,
+        project_id=app_settings.DY_SIDECAR_PROJECT_ID,
+        action=RabbitEventMessageType.RELOAD_IFRAME,
+    )
+    await _post_rabbit_message(app, message)
 
 
 def setup_rabbitmq(app: FastAPI) -> None:
     async def on_startup() -> None:
-        app.state.rabbitmq = RabbitMQ(app)
-
-        log.info("Connecting to rabbitmq")
-        await app.state.rabbitmq.connect()
-        log.info("Connected to rabbitmq")
+        app_settings: ApplicationSettings = app.state.settings
+        assert app_settings.RABBIT_SETTINGS  # nosec
+        settings = app_settings.RABBIT_SETTINGS
+        await wait_till_rabbitmq_responsive(settings.dsn)
+        with log_context(log, logging.INFO, msg="Create RabbitMQClient"):
+            app.state.rabbitmq_client = RabbitMQClient(
+                client_name=f"dynamic-sidecar_{app_settings.DY_SIDECAR_NODE_ID}",
+                settings=settings,
+            )
 
     async def on_shutdown() -> None:
-        if app.state.background_log_fetcher is None:
-            log.warning("No rabbitmq to close")
-            return
-
-        await app.state.rabbitmq.close()
-        log.info("stopped rabbitmq")
+        if app.state.rabbitmq_client:
+            await app.state.rabbitmq_client.close()
```

Comment:
This is relative to resources occupied by this service. Disconnections if this client dies unexpectedly are already handled by the rabbit server.

Reply:
so the rabbitMQ **client** would disconnect in case of power loss... actually the client/server in rabbit are running heartbeats. So there is no issue with this.